backend/src/db-utils.js

```diff
@@ -149,14 +158,38 @@ const queryDashboardBlocksAndTxs = async (from_indexer = false) => {
       from_indexer
     ),
   ]);
+  if (from_indexer) {
+    await Promise.all(
+      transactions.map(async (transaction) => {
+        const actions = await queryRows(
+          [
+            `SELECT transaction_hash, index, action_kind as kind, args as args
+          FROM transaction_actions
+          WHERE transaction_hash = :hash
+          ORDER BY index`,
+            {
+              hash: transaction.hash,
+            },
+          ],
+          from_indexer
+        );
+        transaction.actions = actions.map((action) => {
+          return {
+            kind: action.kind,
+            args: JSON.parse(action.args),
+          };
+        });
+      })
+    );
+  }
   await Promise.all(
     transactions.map(async (transaction) => {
       const actions = await queryRows(
         [
           `SELECT transaction_hash, action_index, action_type as kind, action_args as args
-      FROM actions
-      WHERE transaction_hash = :hash
-      ORDER BY action_index`,
+        FROM actions
+        WHERE transaction_hash = :hash
+        ORDER BY action_index`,
```

Comment:
Apply the same suggestion I had above: https://github.com/near/near-explorer/pull/461#discussion_r515196362

Avoid copy-pasting as it is really getting hard to maintain. Here you have a bug now since if `from_indexer` is true, then you will fetch the actions from the indexer DB first, and then from the old DB right after that.

P.S. Also, you can re-write the `map` of queries (it is almost always a bad idea to do queries in a loop) into a single query which selects `where transaction_hash IN :transaction_hashes` (see [`IN` operator](https://www.w3schools.com/sql/sql_in.asp))